File: app/src/main/java/closer/vlllage/com/closer/api/models/PinResult.java
package closer.vlllage.com.closer.api.models;

import closer.vlllage.com.closer.store.models.Pin;

public class PinResult extends ModelResult {
    public String from;
    public String to;
    public GroupMessageResult message;

    public static Pin from(PinResult pinResult) {
        Pin pin = new Pin();
        pin.setId(pinResult.id);
        updateFrom(pin, pinResult);
        return pin;
    }

    public static Pin updateFrom(Pin pin, PinResult pinResult) {
        pin.setFrom(pinResult.from);
        pin.setTo(pinResult.to);
        return pin;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/PinResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.store.models.Pin

class PinResult : ModelResult() {
    var from: String? = null
    var to: String? = null
    var message: GroupMessageResult? = null

    companion object {

        fun from(pinResult: PinResult): Pin {
            val pin = Pin()
            pin.id = pinResult.id
            updateFrom(pin, pinResult)
            return pin
        }

        fun updateFrom(pin: Pin, pinResult: PinResult): Pin {
            pin.from = pinResult.from
            pin.to = pinResult.to
            return pin
        }
    }
}
